core/src/main/clojure/xtdb/object_store.clj

```diff
@@ -61,118 +52,3 @@
 
 (defn obj-missing-exception [k]
   (IllegalStateException. (format "Object '%s' doesn't exist." k)))
-
-(defn get-path ^java.nio.ByteBuffer [^Path root-path, ^String k]
-  (let [from-path (.resolve root-path k)]
-    (when-not (util/path-exists from-path)
-      (throw (obj-missing-exception k)))
-
-    (util/->mmap-path from-path)))
-
-(defn get-path-range ^java.nio.ByteBuffer [^Path root-path, ^String k, ^long start, ^long len]
-  (ensure-shared-range-oob-behaviour start len)
-
-  (let [from-path (.resolve root-path k)]
-    (when-not (util/path-exists from-path)
-      (throw (obj-missing-exception k)))
-
-    (with-open [in (util/->file-channel from-path #{:read})]
-      (.map in FileChannel$MapMode/READ_ONLY start (max 1 (min (- (.size in) start) len))))))
-
-(defn put-path [^Path root-path, ^String k, ^ByteBuffer buf]
-  (let [buf (.duplicate buf)
-        to-path (.resolve root-path k)]
-    (util/mkdirs (.getParent to-path))
-
-    (if (identical? (FileSystems/getDefault) (.getFileSystem to-path))
-      (if (util/path-exists to-path)
-        to-path
-        (util/write-buffer-to-path-atomically buf root-path to-path))
-
-      (util/write-buffer-to-path buf to-path))))
-
-(defn list-path
-  ([^Path root-path]
-   (with-open [dir-stream (Files/walk root-path (make-array FileVisitOption 0))]
-     (vec (sort (for [^Path path (iterator-seq (.iterator dir-stream))
-                      :when (Files/isRegularFile path (make-array LinkOption 0))]
-                  (str (.relativize root-path path)))))))
-
-  ([^Path root-path, ^String dir]
-   (let [dir (.resolve root-path dir)]
-     (when (Files/exists dir (make-array LinkOption 0))
-       (with-open [dir-stream (Files/newDirectoryStream dir)]
-         (vec (sort (for [^Path path dir-stream]
-                      (str (.relativize root-path path))))))))))
-
-(deftype FileSystemObjectStore [^Path root-path, ^ExecutorService pool]
-  ObjectStore
-  (getObject [_this k] (CompletableFuture/completedFuture (get-path root-path k)))
-
-  (getObjectRange [_this k start len]
-    (CompletableFuture/completedFuture
-     (get-path-range root-path k start len)))
-
-  (getObject [_this k out-path]
-    (CompletableFuture/supplyAsync
-     (reify Supplier
-       (get [_]
-         (let [from-path (.resolve root-path k)]
-           (when-not (util/path-exists from-path)
-             (throw (obj-missing-exception k)))
-
-           (when-not (util/path-exists out-path)
-             (util/copy-file-atomically root-path from-path out-path))
-
-           out-path)))))
-
-  (putObject [_this k buf] (util/completable-future pool (put-path root-path k buf)))
-  (listObjects [_this] (list-path root-path))
-  (listObjects [_this dir] (list-path root-path dir))
-  (deleteObject [_this k] (util/completable-future pool (util/delete-file (.resolve root-path k))))
-
-  Closeable
-  (close [_this]
-    (util/shutdown-pool pool)))
-
-(derive ::file-system-object-store :xtdb/object-store)
```

Comment:
I think the reason we kept the file-system object store was for remote buffer pool + remote file stores?

Reply:
I'm not sure about this - won't the networked obj-store behave differently from the local BP? e.g. local BP knows it's entirely in control of the directory, networked obj-store will likely need to do some kind of put-to-a-temp-file then copy dance?

Reply:
I don't think it will behave differently, we make it a requirement of a local bp to be able to share a data-dir with other processes. The acceptance tests will then include the concurrent checks over a networked filesystem.

need for atomicity of write (never seeing partial files, or making them discoverable), and the immutability property are our friends.

Also not a problem to revert a delete if the team is still not sure.

Reply:
A specialisation may be possible for performance to have a single process directory with a lock so we can manage more in memory - however, this is not how the current disk buffer pool works, it defers to the disk for listings and so on.

I would add this specialisation if the one multi-process supporting buffer pool is not sufficient.